include/fmt/std.h

```diff
@@ -416,6 +416,95 @@ template <typename Char> struct formatter<std::error_code, Char> {
   }
 };
 
+#if FMT_USE_RTTI
```

Comment:
Let's move this check to the implementation of `write_demangled_name` and reduce conditional compilation at call sites.

Reply:
Conditional compilation at call site is protect call ``typeid`` if RTTI is disabled.
I think we can't reduce usage of conditional compilation.

Reply:
The `FMT_USE_RTTI` guard guards the `write_demangled_name` but also `formatter<std::type_info, Char >`.

Reply:
And the 2nd one guards the use of `typeid` inside the exception formatter.